### DISCUSSION TITLE
Tasmota Support for RTC Chip PCF8563T

### DISCUSSION THREAD

**Discussion by jkwim** [arendst/Tasmota#20422] [Q&A] [answered]
I have the following RTC Clock + ESP8266 module.
It has PCF8563T as the real time clock chip.

![image](https://github.com/arendst/Tasmota/assets/84637540/9d384910-4ca8-48e0-9281-f1c9d2b5e095)
[https://www.aliexpress.com/item/1005001620047697.html](https://www.aliexpress.com/item/1005001620047697.html)

I know that Tasmota has support for DS3231 chip.

PCF8563T also works with I2C.

I could not find any info on whether Tasmota has drivers for PCF8563T though the following discussion come to a close match:

[Trying to Tasmotize Ubibot WS1 #17336](https://github.com/arendst/Tasmota/discussions/17336)

Can somebody enlighten me whether any progress has been made to incorporate drivers for PCF8563T?

**Comment by barbudor** [comment]
PCF85363 is supported since Apr 2022 way before the thread you are refering
You need to compile `tasmota` using the following lines in your `user_config_override.h`
```
#USE_RTC_CHIPS
#USE_PCF85363
```

**Reply by jkwim** [reply]
That's great! Serves my purpose!

Thanks!

**Reply by jkwim** [reply]
I compiled in:
```
        #ifndef USE_DISPLAY
          #define USE_DISPLAY
        #endif
        #ifndef USE_I2C
          #define USE_I2C
        #endif
        #ifndef USE_DISPLAY_SSD1306
          #define USE_DISPLAY_SSD1306
        #endif

        #define USE_RTC_CHIPS
        #define USE_PCF85363

        #define RTC_NTP_SERVER
```

This is the physical connection:
![image](https://github.com/arendst/Tasmota/assets/84637540/4d0309b6-ff2c-47d6-958f-ddd19c042e5a)

Tasmota config:
![image](https://github.com/arendst/Tasmota/assets/84637540/666b751c-add6-4d24-8f35-de06677d9f49)

When loaded I see the Display getting detected but no message for detection of PCF8563.

```
00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at F6, Count 22
00:00:00.055 QPC: Count 1
00:00:00.065 I2C: SSD1306 found at 0x3c
00:00:00.093 DSP: SSD1306
00:00:00.098 Project tasmota - Tasmota Version 12.3.1.3(dspl-rtc-ntp)-2_7_4_9(2024-01-08T20:19:26)
00:00:00.555 WIF: Connecting to AP1 xxxxxx Channel 4 BSSId 98:A9:42:1F:49:2E in mode 11n as tasmota-64C246-0582...
00:00:03.759 WIF: Connected
00:00:04.012 HTP: Web server active on tasmota-64C246-0582 with IP address 192.168.1.108
17:09:20.037 RSL: INFO1 = {"Info1":{"Module":"WiFi RTC Clock","Version":"12.3.1.3(dspl-rtc-ntp)","FallbackTopic":"cmnd/DVES_64C246_fb/","GroupTopic":"cmnd/tasmotas/"}}
17:09:20.043 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-64C246-0582","IPAddress":"192.168.1.108"}}
17:09:20.053 RSL: INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":12}}
17:09:20.062 RSL: RESULT = {"POWER":"ON"}
17:09:20.065 RSL: POWER = ON
17:09:21.435 QPC: Reset
17:09:24.391 RSL: STATE = {"Time":"2024-01-08T17:09:24","Uptime":"0T00:00:10","UptimeSec":10,"Heap":25,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":0,"POWER":"ON","Wifi":{"AP":1,"SSId":"xxxxxx","BSSId":"98:A9:42:1F:49:2E","Channel":4,"Mode":"11n","RSSI":62,"Signal":-69,"LinkCount":1,"Downtime":"0T00:00:05"}}

```

Am I supposed to see any message similar to:

`00:00:00.065 I2C: SSD1306 found at 0x3c`

**Reply by jkwim** [reply]
From Release Notes:
```
Apr 13, 2022
Tasmota v11.1.0 Ostara
Support for PCF85363 RTC as used in Shelly 3EM #13515

Consolidate three RTC chip drivers (DS3231, BM8563, PCF85363) into one driver updating RTC as soon as possible after restart
DS3231 I2C address define USE_RTC_ADDR into DS3231_ADDRESS
```

I am on 12.3.1.3 so I should be good for compiling the option in.

**Reply by barbudor** [reply]
Your GPIO configuration is wrong*
You can have only 1x I2C bus and every I2C devices should be connected to the bus


*on ESP32, tasmota32 as limited support for 2 buses but only a couple of drivers have been modified to support it

**Reply by sfromis** [reply]
On ESP8266, you cannot have two I2C busses. You need to connect both the display and sensor on the same I2C pins.

**Reply by jkwim** [reply]
Thanks.

Well I have a strange piece of hardware.

If I trace the connections using continuity checks, my original diagram stands correct:

![image](https://github.com/arendst/Tasmota/assets/84637540/eccf7ab5-b65b-4a6e-ae22-e47d80b875cd)

Hardware looks like this:
![image](https://github.com/arendst/Tasmota/assets/84637540/651f093d-6532-48a0-a53b-42d0a2707f3c)
![image](https://github.com/arendst/Tasmota/assets/84637540/cda6f6ac-cfde-48eb-bd3b-e02521eb9d37)

![image](https://github.com/arendst/Tasmota/assets/84637540/1453a1f9-c1d8-4c6d-a828-edcac5992655)

Display is connected to GPIO4 & GPIO5

![image](https://github.com/arendst/Tasmota/assets/84637540/d85dc940-b923-414f-9235-36d997caecd1)

Display works fine when I2C bus is defined for these pins.

If I remove this definition and use GPIO12/GPIO14 for the I2C bus then both Display and RTC Chip get detected.
![image](https://github.com/arendst/Tasmota/assets/84637540/2d00e912-10fa-41b0-95e9-98e85816633f)

```
00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at F8, Count 92
00:00:00.055 QPC: Count 1
00:00:00.058 CFG: CR 383/699, Busy 0
00:00:00.061 I2C: PCF85363 found at 0x51
00:00:00.065 ROT: Mode 1
00:00:00.066 I2C: SSD1306 found at 0x3c
00:00:00.070 DSP: SSD1306
00:00:00.072 SRC: Restart
00:00:00.074 Project tasmota - Tasmota Version 12.3.1.3(dspl-rtc-ntp)-2_7_4_9(2024-01-08T20:19:26)
```

However the display does not work in this configuration ie. DisplayText command does nothing.

NTP Syncing works towards RTC Chip:

```
00:00:02.481 NTP: Sync time...
00:00:02.488 WIF: DNS resolved '2.pool.ntp.org' (162.159.200.123) in 6 ms
00:00:02.522 RTC: UTC 2024-01-09T15:46:48, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
21:16:48.001 RTC: Synced by NTP
21:16:48.005 RTC: PCF85363 re-synced (UTC) 2024-01-09T15:46:48

```

However if I remove the internet connection and reboot, time never gets synced back from RTC Chip, but continuous attempts are being made to sync with NTP servers.

```
00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at F4, Count 64
00:00:00.055 QPC: Count 1
00:00:00.058 CFG: CR 383/699, Busy 0
00:00:00.061 I2C: PCF85363 found at 0x51
00:00:00.065 ROT: Mode 1
00:00:00.066 I2C: SSD1306 found at 0x3c
00:00:00.069 DSP: SSD1306
00:00:00.072 SRC: Restart
00:00:00.074 Project tasmota - Tasmota Version 12.3.1.3(dspl-rtc-ntp)-2_7_4_9(2024-01-08T20:19:26)
00:00:00.226 WIF: Checking connection...
00:00:00.227 WIF: Attempting connection...
00:00:00.563 WIF: Connecting to AP1 xxxxxxxxxxxxx Channel 4 BSSId xxxxxxxxxxxxxx in mode 11n as tasmota-64C246-0582...
00:00:01.752 WIF: Checking connection...
00:00:01.753 WIF: Connected
00:00:02.005 HTP: Web server active on tasmota-64C246-0582 with IP address 192.168.1.108
00:00:02.394 WIF: Sending Gratuitous ARP
00:00:02.395 NTP: Sync time...
00:00:02.403 WIF: DNS resolved '2.pool.ntp.org' (2.255.255.255) in 6 ms
00:00:03.409 NTP: No reply from 2.255.255.255
00:00:03.421 NTP: Sync time...
00:00:03.441 WIF: DNS resolved '2.europe.pool.ntp.org' (2.255.255.255) in 18 ms
00:00:04.447 NTP: No reply from 2.255.255.255
00:00:04.455 NTP: Sync time...
00:00:04.462 WIF: DNS resolved '2.nl.pool.ntp.org' (2.255.255.255) in 5 ms
00:00:05.468 NTP: No reply from 2.255.255.255
00:00:05.572 RSL: INFO1 = {"Info1":{"Module":"WiFi RTC Clock","Version":"12.3.1.3(dspl-rtc-ntp)","FallbackTopic":"cmnd/DVES_64C246_fb/","GroupTopic":"cmnd/tasmotas/"}}
00:00:05.578 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-64C246-0582","IPAddress":"192.168.1.108"}}
00:00:05.588 RSL: INFO3 = {"Info3":{"RestartReason":"Software/System restart","BootCount":35}}
00:00:05.597 RSL: RESULT = {"POWER":"ON"}
00:00:05.600 RSL: POWER = ON
00:00:05.603 NTP: Sync time...
00:00:05.610 WIF: DNS resolved '0.pool.ntp.org' (0.255.255.255) in 5 ms
00:00:06.616 NTP: No reply from 0.255.255.255
00:00:06.657 QPC: Reset
```

What could be the reason for this?

(I will park the Display Issue for now. However note  that I have two units of this HW, one still running Factory Firmware. It works fine with both display and RTC Clock working at the same time. When running with the battery, the time gets saved and retrieved successfully. Unfortunately there isn't any detail published by the sellers about the original FW. I do not even know who to change the time.)


Also I noticed that there is a hard coded NTP server (0.pool.ntp.org) apart from the 3 x #defines in  `my_user_config.h`
```

00:00:05.603 NTP: Sync time...
00:00:05.610 WIF: DNS resolved '0.pool.ntp.org' (0.255.255.255) in 5 ms
00:00:06.616 NTP: No reply from 0.255.255.255
```

**Reply by arendst** [reply]
If all's well expect a response like this:

```
00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at F9, Count 1355
00:00:00.055 QPC: Count 1
00:00:00.060 I2C: PCF85363 found at 0x51
00:00:00.063 RTC: UTC 2024-01-09T16:43:50Z, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
17:43:50.000 RTC: Synced by PCF85363
17:43:50.005 ROT: Mode 1
17:43:50.006 I2C: ADE7880 found at 0x38
17:43:50.085 NRG: Init driver 23
17:43:50.086 SRC: Restart
17:43:50.087 Project tasmota - Shelly 3EM Version 13.3.0.2(theo)-2_7_4_9(2024-01-09T17:37:54)
17:43:50.142 WIF: Attempting connection...
17:43:51.001 WIF: Connecting to AP1 indebuurt_IoT Channel 11 BSSId 18:E8:29:CA:17:C1 in mode 11n as shelly3em...
17:43:52.522 WIF: Connected
17:43:52.774 HTP: Web server active on shelly3em with IP address 192.168.2.222
17:43:53.266 RTC: UTC 2024-01-09T16:40:41Z, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
17:40:41.000 RTC: Synced by NTP
17:40:41.004 RTC: PCF85363 re-synced (UTC) 2024-01-09T16:40:41Z
17:40:41.506 MQT: Attempting connection...
17:40:41.519 MQT: Connected
```

In this case the Shelly EM3 was pulled from the bin hence the time difference between the battery powered PCF85363 and NTP. NTP synced the PCF85363 and on next power cycle it reports near realtime.

```
00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at F4, Count 1360
00:00:00.056 QPC: Count 1
00:00:00.060 I2C: PCF85363 found at 0x51
00:00:00.063 RTC: UTC 2024-01-09T16:48:22Z, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
17:48:22.000 RTC: Synced by PCF85363
17:48:22.005 ROT: Mode 1
17:48:22.006 I2C: ADE7880 found at 0x38
17:48:22.084 NRG: Init driver 23
17:48:22.085 SRC: Restart
17:48:22.087 Project tasmota - Shelly 3EM Version 13.3.0.2(theo)-2_7_4_9(2024-01-09T17:37:54)
17:48:22.141 WIF: Attempting connection...
17:48:23.001 WIF: Connecting to AP1 indebuurt_IoT Channel 11 BSSId 18:E8:29:CA:17:C1 in mode 11n as shelly3em...
17:48:24.526 WIF: Connected
17:48:24.779 HTP: Web server active on shelly3em with IP address 192.168.2.222
17:48:25.249 RTC: UTC 2024-01-09T16:48:26Z, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
17:48:26.000 RTC: Synced by NTP
17:48:26.530 MQT: Attempting connection...
17:48:26.550 MQT: Connected
```

**Reply by sfromis** [reply]
To get around the issue with the hardware seeing I2C as two busses, I'd add wires connecting gpio 4/5 to 14/12 respectively (short wires across the ESP-12E module, or what works best for you), and leave the latter two unconfigured. The I2C bus is designed to serve multiple devices on the same two pins, instead of multiple busses, and this is what Tasmota was designed for too.

**Comment by jkwim** [comment]
> To get around the issue with the hardware seeing I2C as two busses, I'd add wires connecting gpio 4/5 to 14/12 respectively (short wires across the ESP-12E module, or what works best for you), and leave the latter two unconfigured. The I2C bus is designed to serve multiple devices on the same two pins, instead of multiple busses, and this is what Tasmota was designed for too.

OK. Something that I learned new. 

The following messages indicate the driver presence but not the presence of the actual device. I physically disconnected the display but could see the messages. That eliminated part of the confusion that I had as to how the display get detected when it is connected to a different set of pins.

```
00:00:00.061 I2C: PCF85363 found at 0x51
00:00:00.064 ROT: Mode 1
00:00:00.066 I2C: SSD1306 found at 0x3c
00:00:00.069 DSP: SSD1306
```

Next I will try soldering the wires as suggested to get the display working.

**Reply by jkwim** [reply]
After flashing the second device configuring only 1x I2C bus, I have another confusion.

The first unit which was originally configured with 2 x I2C busses detected the display only at first. Then I removed the GPIO4/5 I2C config and then it was showing two drivers getting detected.

```
00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at F5, Count 114
00:00:00.055 QPC: Count 1
00:00:00.057 CFG: CR 383/699, Busy 0
00:00:00.062 I2C: BM8563 found at 0x51
00:00:00.065 RTC: UTC 2024-01-11T15:16:28, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
20:46:28.000 RTC: Synced by BM8563
20:46:28.003 ROT: Mode 1
20:46:28.005 I2C: SSD1306 found at 0x3c
20:46:28.009 DSP: SSD1306
```

On the second unit through I have the display drivers compiled in, I defined only the GPIO12/14 as I2C for the RTC.
In this unit I do not see any display driver getting detected.

```
00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at F8, Count 20
00:00:00.055 QPC: Count 1
00:00:00.057 CFG: CR 385/699, Busy 0
00:00:00.061 I2C: PCF85363 found at 0x51
00:00:00.064 ROT: Mode 1
00:00:00.067 SRC: Restart
```
[Note that I have not flashed the new firmware with BM8563 driver in to this device]

Wonder whether some kind of caching happens when you try to detect a driver and then the GPIO's are released, the previously detected device driver still shows up.

**Reply by sfromis** [reply]
There is no caching at all. When configuring I2C pins twice, only the first set is used, the 2nd is just ignored. When you see both BM8663 and SSD1306 on the "first unit" with I2C on gpio 12/14, are they both working? If yes, that would imply that they are already wired up reasonably, without needing the suggested extra wiring, which may still be relevant for the "second unit".

**Comment by jkwim** [comment]
> If all's well expect a response like this:
> 
> ```
> 00:00:00.001 HDW: ESP8266EX
> 00:00:00.050 CFG: Loaded from flash at F9, Count 1355
> 00:00:00.055 QPC: Count 1
> 00:00:00.060 I2C: PCF85363 found at 0x51
> 00:00:00.063 RTC: UTC 2024-01-09T16:43:50Z, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
> 17:43:50.000 RTC: Synced by PCF85363
> 17:43:50.005 ROT: Mode 1
> 17:43:50.006 I2C: ADE7880 found at 0x38
> 17:43:50.085 NRG: Init driver 23
> 17:43:50.086 SRC: Restart
> 17:43:50.087 Project tasmota - Shelly 3EM Version 13.3.0.2(theo)-2_7_4_9(2024-01-09T17:37:54)
> 17:43:50.142 WIF: Attempting connection...
> 17:43:51.001 WIF: Connecting to AP1 indebuurt_IoT Channel 11 BSSId 18:E8:29:CA:17:C1 in mode 11n as shelly3em...
> 17:43:52.522 WIF: Connected
> 17:43:52.774 HTP: Web server active on shelly3em with IP address 192.168.2.222
> 17:43:53.266 RTC: UTC 2024-01-09T16:40:41Z, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
> 17:40:41.000 RTC: Synced by NTP
> 17:40:41.004 RTC: PCF85363 re-synced (UTC) 2024-01-09T16:40:41Z
> 17:40:41.506 MQT: Attempting connection...
> 17:40:41.519 MQT: Connected
> ```
> 
> In this case the Shelly EM3 was pulled from the bin hence the time difference between the battery powered PCF85363 and NTP. NTP synced the PCF85363 and on next power cycle it reports near realtime.
> 
> ```
> 00:00:00.001 HDW: ESP8266EX
> 00:00:00.050 CFG: Loaded from flash at F4, Count 1360
> 00:00:00.056 QPC: Count 1
> 00:00:00.060 I2C: PCF85363 found at 0x51
> 00:00:00.063 RTC: UTC 2024-01-09T16:48:22Z, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
> 17:48:22.000 RTC: Synced by PCF85363
> 17:48:22.005 ROT: Mode 1
> 17:48:22.006 I2C: ADE7880 found at 0x38
> 17:48:22.084 NRG: Init driver 23
> 17:48:22.085 SRC: Restart
> 17:48:22.087 Project tasmota - Shelly 3EM Version 13.3.0.2(theo)-2_7_4_9(2024-01-09T17:37:54)
> 17:48:22.141 WIF: Attempting connection...
> 17:48:23.001 WIF: Connecting to AP1 indebuurt_IoT Channel 11 BSSId 18:E8:29:CA:17:C1 in mode 11n as shelly3em...
> 17:48:24.526 WIF: Connected
> 17:48:24.779 HTP: Web server active on shelly3em with IP address 192.168.2.222
> 17:48:25.249 RTC: UTC 2024-01-09T16:48:26Z, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
> 17:48:26.000 RTC: Synced by NTP
> 17:48:26.530 MQT: Attempting connection...
> 17:48:26.550 MQT: Connected
> ```

The way I simulated loss of internet was to remove the WAN link from the WiFi router.

However the router's cache may be giving out the DNS resolution.

When this happens I do not see the RTC Clock helping to to get the time synced. All I see is repeated attempts to sync via internet and the syncing resumes when WAN link is restored. In between no attempt is being made to sync from RTC.

I do not see RTC driver kicking in immediately after boot like in the example above.

![image](https://github.com/arendst/Tasmota/assets/84637540/f4caffaa-92b9-4f6b-a542-32e003e7b308)

**Reply by sfromis** [reply]
In your colorful screenshots I do not see the relevant test case, which would be to first boot with both PCF85363 and NTP server access. This will allow Tasmota to synchronize the PCF85363 with the real time from the NTP server. Only near the end there is the important message about `RTC: PCF85363 re-synced`, but that's where the testing should *start*.

When this has been done, I'd expect the **next** boot to be able to get time from the PCF85363 in the same way as in the log clips posted by arendst.

**Reply by jkwim** [reply]
It is in fact the log after a subsequent reboot.

Here I tested again.

When internet is there:
```

00:00:02.406 NTP: Sync time...
00:00:02.447 WIF: DNS resolved '2.pool.ntp.org' (162.159.200.1) in 39 ms
00:00:02.480 RTC: UTC 2024-01-10T00:09:46, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
05:39:46.000 RTC: Synced by NTP
05:39:46.004 RTC: PCF85363 re-synced (UTC) 2024-01-10T00:09:46
```

Afterwards I disconnected internet and rebooted:

```
00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at F6, Count 102
00:00:00.056 QPC: Count 1
00:00:00.058 CFG: CR 383/699, Busy 0
00:00:00.061 I2C: PCF85363 found at 0x51
00:00:00.065 ROT: Mode 1
00:00:00.066 I2C: SSD1306 found at 0x3c
00:00:00.070 DSP: SSD1306
00:00:00.072 SRC: Restart
00:00:00.074 Project tasmota - Tasmota Version 12.3.1.3(dspl-rtc-ntp)-2_7_4_9(2024-01-08T20:19:26)
00:00:00.228 WIF: Checking connection...
00:00:00.228 WIF: Attempting connection...
00:00:00.565 WIF: Connecting to AP1 Kalimanku 4G SLT Channel 4 BSSId 98:A9:42:1F:49:2E in mode 11n as tasmota-64C246-0582...
00:00:01.757 WIF: Checking connection...
00:00:01.758 WIF: Connected
00:00:02.009 HTP: Web server active on tasmota-64C246-0582 with IP address 192.168.1.108
00:00:02.409 WIF: Sending Gratuitous ARP
00:00:02.411 NTP: Sync time...
00:00:02.446 WIF: DNS resolved '2.pool.ntp.org' (2.255.255.255) in 34 ms
00:00:03.451 NTP: No reply from 2.255.255.255  
```

It continues to look for NTP server but no attempt is made to sync from RTC

**Reply by sfromis** [reply]
First test case could have been to reboot without disconnecting Internet, and verify whether it was actually able to get time from the PCF85363. The log suggests that the answer would be no, meaning that having Internet connection or not is likely not part of the case.

Just like the log clips posted by arendst, where the 2nd boot first gets time from the PCF85363 and later synchronizes with NTP getting same time (or very close) again.

Anyway, the log you posted now does indeed suggest failure to get the time from the PCF85363 after the boot. I do not know much about the chip, but your images suggest that you did not have a battery installed. Just pure speculation, but maybe it only provides time if having battery power?

**Comment by jkwim** [comment]
> First test case could have been to reboot without disconnecting Internet, and verify whether it was actually able to get time from the PCF85363. The log suggests that the answer would be no, meaning that having Internet connection or not is likely not part of the case.
> 
> Just like the log clips posted by arendst, where the 2nd boot first gets time from the PCF85363 and later synchronizes with NTP getting same time (or very close) again.
> 
> Anyway, log you posted now does indeed suggest failure to get the time from the PCF85363 after the boot. I do not know much about the chip, but your images suggest that you did not have a battery installed. Just pure speculation, but maybe it only provides time if having battery power?

I have a battery installed. The battery directly supplies power to the chip via a diode (reverse polarity protection and voltage drop I suppose).

Anyway I have two units. I will flash the second one also and test in case if the HW is faulty. The default firmware in the unit works fine even with the separate connections of RTC and Display. May be the code changes the pin definition dynamically. 

Without the battery it always starts time with 00:00 but once the battery is installed, it loads correct time after a powered down period.

![image](https://github.com/arendst/Tasmota/assets/84637540/0c3e369b-96f7-4db2-b178-f22e4c76b492)
![image](https://github.com/arendst/Tasmota/assets/84637540/ec977b8f-a1ec-4db8-a13f-c9da86fe8412)

Although the product is called "WiFi Clock" sellers (all the sellers that I checked) have no instructions published for operation of the device or details about original code. So I have to flash them for any usage of mine (either write my own code or Tasmota)

**Comment by arendst** [comment]
Looking at your pictures, what made you think you have a PCF85363? From what I see it looks more like a BM8563 also supported by Tasmota.

**Reply by jkwim** [reply]
> Looking at your pictures, what made you think you have a PCF85363? From what I see it looks more like a BM8563 also supported by Tasmota.

Sellers specification says:
0.96 inch OLED display: the driver IC is SSD1306; 
ESP12FWF module: FLASH capacity is 4 MBYTE; 
MICRO USB5V power supply interface; 
PCF8563T clock chip; 
CR1220 button battery; supply power to clock circuit after power off 

Also the datasheet for PCF8563 has a model PCF8563T which looks like the CHIP I have on the device.

![image](https://github.com/arendst/Tasmota/assets/84637540/07f8512a-b065-4b61-90d4-fa710c3332c6)
![image](https://github.com/arendst/Tasmota/assets/84637540/77c40dbe-1225-40ec-88bb-9f0990bc9ffd)

All of the above took me to assume that it is a PCF8563.

I looked at the Datasheet of BM8563 and pin configuration is identical.

![image](https://github.com/arendst/Tasmota/assets/84637540/cc6f6ed4-5054-4455-9b9a-31dc585bf1f3)

**Reply by jkwim** [reply]
FYI, I flashed the second unit also and configured only GPIO12/14.

Same observation as first unit. So that confirms that hardware is ok.

During boot PCF8563 does not get accessed.

**Reply by arendst** [reply]
Again, the issue is not BM or PCF. The issue is 8563 vs 85363!!!

So you need USE_BM8563

**Reply by barbudor** [reply]
I'm responsible as I overlooked the question and pointed to USE_PCF85363 

Sorry @jkwim @arendst 
Shame on me

**Reply by jkwim** [reply, marked as answer]
Finally!
1. Synced via internet
2. Rebooted
```
00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at FA, Count 109
00:00:00.055 QPC: Count 1
00:00:00.058 CFG: CR 383/699, Busy 0
00:00:00.062 I2C: BM8563 found at 0x51
00:00:00.066 RTC: UTC 2024-01-11T14:47:59, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
20:17:59.000 RTC: Synced by BM8563
20:17:59.004 ROT: Mode 1
20:17:59.005 I2C: SSD1306 found at 0x3c
20:17:59.009 DSP: SSD1306
20:17:59.011 SRC: Restart
20:17:59.014 Project tasmota - Tasmota Version 12.3.1.3(dspl-rtc-ntp)-2_7_4_9(2024-01-11T13:37:32)
.
.
.
20:18:03.414 NTP: Sync time...
20:18:03.458 WIF: DNS resolved '2.pool.ntp.org' (162.159.200.1) in 42 ms
20:18:03.501 RTC: UTC 2024-01-11T14:48:03, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
20:18:03.000 RTC: Synced by NTP
```

3. Rebooted after disconnecting internet
```
00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at F9, Count 110
00:00:00.056 QPC: Count 1
00:00:00.058 CFG: CR 383/699, Busy 0
00:00:00.063 I2C: BM8563 found at 0x51
00:00:00.066 RTC: UTC 2024-01-11T14:55:05, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
20:25:05.000 RTC: Synced by BM8563
```

4. Internet reconnected
```
20:27:21.415 NTP: Sync time...
20:27:21.494 WIF: DNS resolved '2.nl.pool.ntp.org' (162.159.200.123) in 77 ms
20:27:21.528 RTC: UTC 2024-01-11T14:57:22, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
20:27:22.000 RTC: Synced by NTP
```

THANK YOU ALL!


PS:
My final `user_config_override.h`:
```
        #ifndef USE_DISPLAY
          #define USE_DISPLAY
        #endif
        #ifndef USE_I2C
          #define USE_I2C
        #endif
        #ifndef USE_DISPLAY_SSD1306
          #define USE_DISPLAY_SSD1306
        #endif

        #define USE_RTC_CHIPS
        //#define USE_PCF85363
        #define USE_BM8563

        #define RTC_NTP_SERVER
```

**Comment by jkwim** [comment]
> I'm responsible as I overlooked the question and pointed to USE_PCF85363
> 
> Sorry @jkwim @arendst Shame on me

No issue. "To err is human!".

Thanks @arendst. Even I was dumb not being able to distinguish the difference.

I will try the suggestion.

Thanks for being with me all this long. I am also learning new things.

I will let you know how it goes.

I want to configure this device as a NTP server for few Tasmota devices which use some timers where internet connection will not be available except when some manual intervention is made.

**Reply by Jason2866** [reply]
Another option for a total independent NTP time source of wifi is adding a cheap GPS receiver to Tasmota. This way you get a high precise NTP time source.
https://tasmota.github.io/docs/GPS-NTP-server/

**Reply by jkwim** [reply]
Thanks. I have seen that. May be my next testing attempt once I acquire a GPS module.

**Comment by jkwim** [comment]
**Enabling NTP Server**

Just for completeness of the post, I am adding the steps that I followed in enabling the NTP server.

1. Following needs to be compiled in:
` #define RTC_NTP_SERVER`
 
 I have two devices running Tasmota 12.3.1.3.
```
 IP .108 running the NTP Server
 IP .114 act as NTP Client
```
 
 2. Enabled NTP Server in .108
 (ignore the time stamp here as it is an old capture)

`RtcNtpServer 1
`
```
23:38:16.922 CMD: RtcNtpServer 1
23:38:16.924 SRC: WebConsole from 192.168.1.118
23:38:16.927 CMD: Grp 0, Cmd 'RTCNTPSERVER', Idx 1, Len 1, Pld 1, Data '1'
23:38:16.933 RSL: RESULT = {"RtcNtpServer":"ON"}
23:38:17.199 CFG: Saved to flash at F7, Count 120, Bytes 4096
23:38:17.435 RTC: NTP server started
```

3. Configured .114 to use .108 as the NTP Server

`ntpserver1 192.168.1.108
`
```
09:22:15.805 CMD: ntpserver1 192.168.1.108
09:22:15.807 SRC: WebConsole from 192.168.1.118
09:22:15.810 CMD: Grp 0, Cmd 'NTPSERVER', Idx 1, Len 13, Pld 192, Data '192.168.1.108'
09:22:15.813 CFG: CR 384/699, Busy 0
09:22:15.815 CFG: CR 384/699, Busy 0
09:22:15.818 RSL: RESULT = {"NtpServer1":"192.168.1.108"}
09:22:16.093 CFG: Saved to flash at FB, Count 37, Bytes 4096
09:22:16.393 NTP: Sync time...
09:22:16.395 WIF: DNS resolved '192.168.1.108' (192.168.1.108) in 0 ms
09:22:16.822 RTC: UTC 2024-01-15T03:52:17, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
09:22:17.001 RTC: Synced by NTP
```

5. Rebooted .114
```
00:00:00.001 HDW: ESP8266EX
00:00:00.050 CFG: Loaded from flash at F7, Count 41
00:00:00.055 QPC: Count 1
00:00:00.058 CFG: CR 384/699, Busy 0
00:00:00.063 I2C: BM8563 found at 0x51
00:00:00.066 RTC: UTC 2024-01-15T04:10:22, DST 2024-03-31T02:00:00, STD 2024-10-27T03:00:00
09:40:22.000 RTC: Synced by BM8563
09:40:22.004 ROT: Mode 1
09:40:22.005 I2C: SSD1306 found at 0x3c
09:40:22.034 DSP: SSD1306
09:40:22.036 SRC: Restart
09:40:22.039 Project tasmota - Tasmota Version 12.3.1.3(dspl-rtc-ntp)-2_7_4_9(2024-01-11T13:37:32)
09:23:13.141 WIF: Checking connection...
09:23:13.142 WIF: Attempting connection...
09:23:13.478 WIF: Connecting to AP1 xx Channel 6 BSSId xxx in mode 11n as tasmota-92A1A5-0421...
09:23:14.004 DBG: [Version 12.3.1.3]
09:23:14.757 WIF: Checking connection...
09:23:14.758 WIF: Connected
09:23:15.010 HTP: Web server active on tasmota-92A1A5-0421 with IP address 192.168.1.114
09:23:15.410 WIF: Sending Gratuitous ARP
09:23:15.411 NTP: Sync time...
09:23:15.412 WIF: DNS resolved '192.168.1.108' (192.168.1.108) in 0 ms
09:23:16.422 NTP: No reply from 192.168.1.108
```

Initially I had a `No reply from 192.168.1.108` error when rebooting. Probably it was my WiFi network. Subsequent reboots produced the desired results.
